fenix/app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -278,6 +279,13 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         window.decorView.layoutDirection = TextUtils.getLayoutDirectionFromLocale(Locale.getDefault())
 
         binding = ActivityHomeBinding.inflate(layoutInflater)
+
+        binding.homeActivityComposeOverlay.setContent {
+            FirefoxTheme {
+                FenixOverlay()
+            }
+        }
```

Comment:
It might run a lower risk of performance impact to observe the data store here and call this whole stack conditionally

Reply:
+1 here

Reply:
I'll look into this. I didn't necessarily want to do a debug settings data store observation at this level, but maybe there's another way to do this. Some initial thoughts could be:
- Have a separate observable boolean `fenixOverlayEnabled` that is driven by a series of other conditions (e.g. `debugDrawerEnabled`)
- Scrap the initial idea of having a universal, global Composable for the entire app and just scope it to the debugging. So make `FenixOverlay` into `DebugOverlay` and then the scoped observation would make more sense.

Reply:
I do think we can probably land something and iterate on it as needed, but the way I'm thinking about it is this:

What if we wanted to use a feature flag or build flag to control whether the overlay was present? For example only including it in debug builds to start. My assumption would be that that would be controlled before the feature package/fragment/whatever got entered, whereas here we are entering the feature first and then determining if we should do anything with it